### PR TITLE
Feature/load score

### PR DESCRIPTION
**Pull request by ChrisBerg**
Added PDF loading to the script page. 
Added the score to the scripting page (below the timeline), placed the timeline and score next to each other.
Also fixed various bugs related to opening a new project, and selecting or creating new scripts and maps.

### CODE REVIEW

File: src/main/webapp/app/map/fabric-map.directive.js [hunk 1]
```diff
@@ -51,16 +51,23 @@
...
+                scope.canvas = canvas;
+
+                var default_bg = "content/images/error.jpg";
                 var error_bg = "content/images/error.jpg";
 
                 function onLoadSuccess(map) {
-                    if(map == ""){
-                        load(default_bg);
-                    } else {
+                    if(map.backgroundImage){
```
Comment (ChrisBerg) [inline src/main/webapp/app/map/fabric-map.directive.js]: This is a way to check whether the map has a background image:
for if(map.backgroundImage)
map.backgroundImage = "" => false
map.backgroundImage = undefined => false
map.backgroundImage = "foo" => true

but for if(map !== "")
map.backgroundImage = "" => false
map.backgroundImage = undefined => true
map.backgroundImage = "foo" => true

it is shorter, too.

File: src/main/java/nl/tudelft/thefirstorder/domain/Script.java [hunk 2]
```diff
@@ -35,59 +34,46 @@
...
-     */
+    @OneToOne(mappedBy = "script")
+    @JsonIgnore
+    private Project project;
+
     public Long getId() {
         return id;
     }
 
-    /**
```
Comment (trombomart) [inline src/main/java/nl/tudelft/thefirstorder/domain/Script.java]: Why did you removed the comments?

File: src/main/java/nl/tudelft/thefirstorder/domain/Script.java [hunk 3]
```diff
@@ -35,59 +34,46 @@
     @Column(name = "name")
     private String name;
 
-    @OneToOne(mappedBy = "script")
-    @JsonIgnore
-    private Project project;
+    @Column(name = "score")
+    private String score;
 
-    @OneToMany(fetch = FetchType.EAGER, mappedBy = "script")
```
Comment (trombomart) [inline src/main/java/nl/tudelft/thefirstorder/domain/Script.java]: Why removed the fetchtype?

File: src/main/webapp/app/entities/script/scripts.html [hunk 4]
```diff
@@ -38,14 +38,12 @@
                                 <span class="hidden-xs hidden-sm"></span>
                             </button>
                             <button type="submit"
-                                    has-any-authority="ROLE_ADMIN.ROLE_DIRECTOR"
```
Comment (trombomart) [inline src/main/webapp/app/entities/script/scripts.html]: These authorities shouldn't be removed

File: src/main/webapp/app/entities/script/scripts.html [hunk 5]
```diff
@@ -38,14 +38,12 @@
...
                                     ui-sref="script.edit({id:script.id})"
                                     class="btn btn-primary btn-sm">
                                 <span class="glyphicon glyphicon-pencil"></span>
                                 <span class="hidden-xs hidden-sm"></span>
                             </button>
                             <button type="submit"
-                                    has-any-authority="ROLE_ADMIN,ROLE_DIRECTOR"
```
Comment (trombomart) [inline src/main/webapp/app/entities/script/scripts.html]: The same for these authorities

File: src/main/webapp/app/map/load-background-dialog.controller.js [hunk 6]
```diff
@@ -61,7 +63,7 @@
 
             vm.map = currentProject.map;
             vm.map.backgroundImage = location;
-            Map.update(vm.map, onUpdateSuccess, onUpdateError);
+            console.log(Map.update(vm.map, onUpdateSuccess, onUpdateError));
```
Comment (trombomart) [inline src/main/webapp/app/map/load-background-dialog.controller.js]: Console log can be removed

File: src/main/webapp/app/scripting/fabric-score.directive.js [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+                var parent = element.parent().parent();
+                var width = parent[0].clientWidth - 30;
+
+                scope.canvas.setWidth(width);
+                scope.canvas.setHeight(width / scope.aspectRatio);
+                scope.canvas.backgroundImage.set({
+                    width: scope.canvas.width,
+                    height: scope.canvas.width / scope.aspectRatio
+                });
+                // scope.canvas.forEachObject(setDrawableProperties);
```
Comment (trombomart) [inline src/main/webapp/app/scripting/fabric-score.directive.js]: This comment can be removed

File: src/main/webapp/index.html [hunk 8]
```diff
@@ -36,6 +35,16 @@
         </div>
     </div>
 
+    <!-- Google Analytics: uncomment and change UA-XXXXX-X to be your site's ID.
```
Comment (trombomart) [inline src/main/webapp/index.html]: Also these comments can be removed

File: src/test/java/nl/tudelft/thefirstorder/web/rest/ScriptResourceIntTest.java [hunk 9]
```diff
@@ -26,14 +29,8 @@
...
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.hamcrest.Matchers.hasItem;
-import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
-import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
-import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
-import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
-import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
-import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
-import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
```
Comment (trombomart) [inline src/test/java/nl/tudelft/thefirstorder/web/rest/ScriptResourceIntTest.java]: Don't use star imports

File: src/test/java/nl/tudelft/thefirstorder/web/rest/ScriptResourceIntTest.java [hunk 10]
```diff
@@ -90,67 +90,16 @@ public void createScript() throws Exception {
         // Create the Script
 
         restScriptMockMvc.perform(post("/api/scripts")
-                .contentType(TestUtil.APPLICATION_JSON_UTF8)
```
Comment (trombomart) [inline src/test/java/nl/tudelft/thefirstorder/web/rest/ScriptResourceIntTest.java]: Why removing these tests?

File: src/test/java/nl/tudelft/thefirstorder/web/rest/UploadResourceIntTest.java [hunk 11]
```diff
@@ -2,29 +2,27 @@
...
 import org.springframework.web.multipart.MultipartFile;
 
 import javax.annotation.PostConstruct;
 import javax.inject.Inject;
-import javax.mail.Multipart;
-
-import java.io.ByteArrayInputStream;
-import java.io.File;
-import java.io.InputStream;
+import java.io.*;
```
Comment (trombomart) [inline src/test/java/nl/tudelft/thefirstorder/web/rest/UploadResourceIntTest.java]: also star import

**Comment by trombomart** [comment]
Works good. Except for some removed code. I think you used the entity generator? This generator removes code we added like the tests etc.

**Comment by ChrisBerg** [comment]
That's right, but i only applied it to the liquibase changelog because my beans were messed up after i added the field manually. I think that the star imports are caused by my auto code format, i'll try to solve that. I don't know for sure why the authorities were removed. I will implement all suggestions from your notes and i will check everything again to make sure nothing else has been changed. Thanks for reviewing!

File: src/main/webapp/app/scripting/fabric-score.directive.js [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+                    });
+                }
+            }
+
+            /**
+             * Resize the canvas based on the timeline.
+             */
+            function resize() {
+                var parent = element.parent().parent();
+                var width = parent[0].clientWidth - 30;
```
Comment (cadw) [inline src/main/webapp/app/scripting/fabric-score.directive.js]: Maybe 30 can be defined as constant, looks like hardcoding.

File: src/main/java/nl/tudelft/thefirstorder/domain/Script.java (hunk 3 above)
Comment (hungnguyen94) [inline src/main/java/nl/tudelft/thefirstorder/domain/Script.java]: `FetchType` should be `EAGER`, but `LAZY` which is the default. I'm not sure why it was set to `EAGER` in the first place actually.

File: src/main/webapp/app/map/fabric-map.directive.js [hunk 13]
```diff
@@ -1,20 +1,14 @@
...
     angular
         .module('thefirstorderApp')
         .directive('fabricMap', fabricMap);
 
-    fabricMap.$inject = ['$window', 'Player', 'Camera', 'Map', 'mapConstants'];
+    fabricMap.$inject = ['$window', 'Player', 'Camera', 'mapConstants', 'ProjectManager', 'Project'];
+
+    function fabricMap($window, Player, Camera, mapConstants, ProjectManager, Project) {
 
-    /**
```
Comment (hungnguyen94) [inline src/main/webapp/app/map/fabric-map.directive.js]: JSDoc shouldn't be removed.

File: src/main/webapp/app/scripting/fabric-score.directive.js [hunk 14]
```diff
@@ -0,0 +1,104 @@
...
+            init();
+            console.log("Score:", scope.score);
+
+            angular.element($window).bind('resize', resize);
+
+            /**
+             * Initializes the script with a canvas and loads the cameras and players.
+             */
+            function init() {
+                Script.get({id: 1}, onLoadSuccess, onLoadError);
```
Comment (hungnguyen94) [inline src/main/webapp/app/scripting/fabric-score.directive.js]: This is a hardcoded script id.

File: src/main/webapp/app/scripting/scripting.html [hunk 15]
```diff
@@ -1,37 +1,63 @@
...
-                        <td>{{vm.selectedCamera.name}}</td>
-                        <td>{{vm.selectedPlayer.name}}</td>
-                    </tr>
+                <tr>
+                    <td>{{vm.selectedCamera.name}}</td>
+                    <td>{{vm.selectedPlayer.name}}</td>
+                </tr>
                 </tbody>
             </table>
+            <canvas fabric-map map="vm.map"
```
Comment (hungnguyen94) [inline src/main/webapp/app/scripting/scripting.html]: I think canvas should be in a separate `div`. Maybe this is why it wasn't correctly responsive?

File: src/test/javascript/karma.conf.js [hunk 16]
```diff
@@ -47,7 +47,6 @@ module.exports = function (config) {
             'src/main/webapp/bower_components/file-saver.js/FileSaver.js',
             'src/main/webapp/bower_components/angular-file-saver/dist/angular-file-saver.bundle.js',
             'src/main/webapp/bower_components/vis/dist/vis.js',
-            'src/main/webapp/bower_components/angular-animate/angular-animate.js',
```
Comment (hungnguyen94) [inline src/test/javascript/karma.conf.js]: This shouldn't be removed.

File: src/main/webapp/index.html [hunk 17]
```diff
@@ -19,7 +19,6 @@
     <link rel="stylesheet" href="content/css/custom.css">
     <link rel="stylesheet" href="content/css/bootstrap-material-design.css" />
     <link rel="stylesheet" href="content/css/ripples.css" />
-    <link rel="stylesheet" href="content/css/vertical-timeline.css" />
```
Comment (hungnguyen94) [inline src/main/webapp/index.html]: This shouldn't be removed.

File: src/main/webapp/app/scripting/fabric-score.directive.js (hunk 14 above)
Comment (ChrisBerg) [inline src/main/webapp/app/scripting/fabric-score.directive.js]: Removing this, using the html5 PDF embed function for showing the script, this has become redundant.

File: src/main/webapp/app/scripting/scripting.html (hunk 15 above)
Comment (ChrisBerg) [inline src/main/webapp/app/scripting/scripting.html]: I believe this actually works, the timeline does not shift over the map. I think i changed this after @rubenwiersma branched.

File: src/main/webapp/app/map/fabric-map.directive.js (hunk 13 above)
Comment (ChrisBerg) [inline src/main/webapp/app/map/fabric-map.directive.js]: Reintroduced it

File: src/main/webapp/app/scripting/fabric-score.directive.js (hunk 12 above)
Comment (ChrisBerg) [inline src/main/webapp/app/scripting/fabric-score.directive.js]: Removed this file

**Pull request merged**